backend/src/db/sql_statement.py

```diff
@@ -0,0 +1,427 @@
+"""Creates SQL statements"""
+
+from enum import Enum, auto
+from db.db_base import DB
+from core.app_logging import getLogger
+from db.SQLFactory import SQLFactory
+from typing import List
+from enum import Enum
+
+LOG = getLogger(__name__)
+
+
+class SQL_data_type(Enum):
+    TEXT = auto()
+    INTEGER = auto()
+    REAL = auto()
+    BLOB = auto()
+
+
+class InvalidSQLStatementException(Exception):
+    """
+    Exception raised when an invalid SQL statement is encountered.
+    """
+    pass
+
+
+class SQL_Executable(object):
+    parent:'SQL_Executable' = None
+
+    def execute(self, params=None, close=False, commit=False):
+        return self.parent.execute(params=params, close=close, commit=commit)
+    
+    def close(self):
+        return self.parent.close()
+    
+    def get_class(self, sql_cls:type)->type:
+        return self.sqlFactory.getClass(sql_cls)
+    
+    @property
+    def sqlFactory(self)->SQLFactory:
+        return self.parent.sqlFactory
+
+
+class SQL(SQL_Executable):
+    """Usage:
+        sql = SQL_statement(db).create_table(
+            "users",
+            [
+                ("name", SQL_data_type.TEXT),
+                ("age",  SQL_data_type.INTEGER)
+            ],
+        ).execute()"""
+
+    def sql(self)->str:
+        if self.sql_statement is None:
+            raise InvalidSQLStatementException("No SQL statement to execute.")
+        return self.sql_statement.sql()
+
+    rslt = None
+    sql_statement = None
```

Comment:
these are in fact instance attributes and should be initialized in `__init__()` and not declared as class attributes: